cinema/views.py

```diff
@@ -136,3 +204,17 @@
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+    def get_permissions(self):
+        if self.action in [
+            "create",
+            "list",
```

Comment:
The 'retrieve' action is missing from the permission check. Consider adding 'retrieve' to the list of actions that allow 'ReadAndCreate' permissions.